rust/main/chains/hyperlane-cosmos/src/libs/account_id_type.rs

```diff
@@ -0,0 +1,40 @@
+use crate::HyperlaneCosmosError;
+
+/// Specifies the account id (address) type
+pub enum AccountIdType {
```

Comment:
nit: We use `strum`'s `EnumString` ([docs](https://docs.rs/strum/latest/strum/derive.EnumString.html#example-howto-use-enumstring)) elsewhere in the code that could be helpful here instead of the manual impls. Example here https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/11e13e56f080e5b533e0fd53b8cddf153a9e85e7/rust/main/hyperlane-core/src/chain.rs#L159

Don't feel strongly though if you'd prefer not to bring that dep into this crate

Reply:
Fixed